utils_cv/tracking/model.py

```diff
@@ -236,113 +309,89 @@ def save(self, path) -> None:
+        result_path = savetxt_results(results, exp_name="single_vid")
         # Save tracking results in tmp
-        mot_accumulator = self.evaluate_mot(gt_root_path, "single_vid", result_file_path)
-        accs.append(mot_accumulator)
-        strsummary = self.mot_summary(accs, ("single_vid",))
+        mot_accumulator = evaluate_mot(gt_root_path, "single_vid", result_path)
+        strsummary = mot_summary([mot_accumulator], ("single_vid",))
         return strsummary
-    
+
+    def eval_mot(self, conf_thres: float, track_buffer: int, im_size: Tuple[int, int], data_root: str,
+                 seqs: list, result_root: str, exp_name: str, if_test_data: bool = False) -> str:
+        """
+        Call the prediction function, saves the tracking results to txt file and provides the evaluation results with motmetrics format.
+        Args:
+            conf_thres: confidence thresh for tracking
+            track_buffer: tracking buffer
+            im_size: image resolution
+            data_root: data root path
+            seqs: list of video sequences subfolder names under MOT challenge data
+            result_root: tracking result path
+            exp_name: experiment name
+            if_test_data: if we evaluate on test set, set True will ignore evaluation
+        Returns:
+            strsummary: str output by method in 'motmetrics' package, containing metrics scores
+        """
+        eval_path = osp.join(result_root, exp_name)
+        if not osp.exists(eval_path):
+            os.makedirs(eval_path)
+        accumulators = []
+        strsummary = None
+        for seq in seqs:
+            image_path = osp.join(data_root, seq, 'img1')
+            result_filename = '{}.txt'.format(seq)
+            result_path = osp.join(result_root, exp_name, result_filename)
+            with open(osp.join(data_root, seq, 'seqinfo.ini')) as seqinfo_file:
+                meta_info = seqinfo_file.read()
+            # framte_rate is set from seqinfo.ini by frameRate
+            frame_rate = int(
+                meta_info[meta_info.find('frameRate') + 10:meta_info.find('\nseqLength')])
+            # If prediction results file exsits, don't have to re-run the prediction
+            if not osp.exists(result_path):
+                # Run tracking.
+                eval_results = self.predict(
+                    image_path,
+                    conf_thres,
+                    track_buffer,
+                    im_size,
+                    frame_rate)
+                result_path = savetxt_results(eval_results, exp_name, result_root,
+                                                   result_filename, if_tmp=False)
+            print(f"Saved tracking results to {result_path}")
+            if not if_test_data:
+                # eval
+                print(f"Evaluate seq: {seq}")
+                mot_accumulator = evaluate_mot(data_root, seq, result_path)
+                accumulators.append(mot_accumulator)
+        if accumulators:
```

Comment:
[minor] change code slightly to improve readability (and remove strsummary = None above)
   if it_test_data:
      return None
   else:
       strsummary = mot_summary(accumulators, seqs)
       return strsummary